chapters/chapter_5/section_5.md

```diff
@@ -239,8 +240,8 @@ dummy_dd_new(GlobalConfig *cfg)
   self->super.worker.disconnect = dummy_dd_disconnect;
   self->super.worker.insert = dummy_worker_insert;
 
-  self->super.format.stats_instance = dummy_dd_format_stats_instance;
-  self->super.format.persist_name = dummy_dd_format_persist_name;
+  self->super.format_stats_instance = dummy_dd_format_stats_instance;
+  self->super.super.super.super.generate_persist_name = dummy_dd_format_persist_name;
```

Comment:
The implementation of this virtual methods has to be changed as well, for example:

https://github.com/balabit/syslog-ng/blob/46e795fd46ea6de4dc78e45a5358dd076252e3ac/modules/http/http.c#L299-L302